tap_mongodb/sync_strategies/oplog.py

```diff
@@ -132,7 +132,7 @@ def sync_collection(client, stream, state, stream_projection, max_oplog_ts=None)
         'ns': {'$eq' : '{}.{}'.format(database_name, collection_name)}
     }
 
-    projection = transform_projection(stream_projection)
+    projection = transform_projection(stream_projection) if stream_projection != {} else None
```

Comment:
Seems a little strange to handle another case of `stream_projection` here when we do already have special cases in `tap_mongodb/__init__.py::load_stream_projection`.

I don't know enough about Mongo to know what is more correct, but I pushed a branch on top of your changes in this branch: https://github.com/singer-io/tap-mongodb/compare/TDL-23609-fix-empty-projection-test...TDL-23609-transform-special-case

If we go with your code, then `projection` is `None`.

If we go with my code, [`projection` is some default projection we hardcoded in the tap](https://github.com/singer-io/tap-mongodb/blob/70adde7927468be4304dbdca635cd884c2350274/tap_mongodb/sync_strategies/oplog.py#L68-L71)

Do you know which is more correct?

Reply:
Ohhh damn, I tried code below which failed the test and keep wondering why its failing.

`    if stream_projection == '' or stream_projection == '""' or not stream_projection or stream_projection == {}:
        return None`

Now I realise that `not stream_projection` evaluates to `True` for empty projection. Your changes work, will push the changes.

Reply:
Pushed the changes.